tokio/src/stream/extend.rs

```diff
@@ -0,0 +1,21 @@
+use crate::stream::{Stream, StreamExt};
+use std::{iter::Extend, marker::Unpin};
+/// Extends a slice from a Stream
+/// # Example
+/// `rust,no_run`
+/// use tokio::stream;
+///
+/// let mut b = vec![-2];
+/// let stream = stream::iter(vec![0, 2, 4, 6]);
+///
+/// stream::extend(&mut buff, s).await;
+/// assert_eq!(vec![-2, 0, 2, 4, 6], buff);
+pub async fn extend<E, S, T>(buff: &mut E, mut b: S)
+where
+    S: Stream<Item = T> + Unpin,
+    E: Extend<T>,
+{
+    while let Some(item) = b.next().await {
+        buff.extend(std::iter::once(item))
+    }
+}
```

Comment:
This can be simplified somewhat.
```suggestion
pub async fn extend<E, S>(buff: &mut E, mut b: S)
where
    S: Stream,
    E: Extend<S::Item>,
{
    crate::pin!(b);
    while let Some(item) = b.next().await {
        buff.extend(std::iter::once(item))
    }
}
```